File: Chapter1_Strategy/MiniDuckSimulator/DuckCaller.java
public class DuckCaller implements QuackBehavior {

  QuackBehavior quackBehavior;

  public DuckCaller() {
    quackBehavior = new Quack();
  }

  public void setQuackBehavior(QuackBehavior qb) {
    quackBehavior = qb;
  }

  public void quack() {
    quackBehavior.quack();
  }
}

File: Chapter1_Strategy/SharpMiniDuckSimulator/DuckCaller.cs
using System;

namespace SharpMiniDuckSimulator
{
    public class DuckCaller : IQuackBehavior
    {

        IQuackBehavior QuackBehavior;

        public DuckCaller()
        {
            QuackBehavior = new Quack();
        }

        public void setQuackBehavior(IQuackBehavior qb)
        {
            QuackBehavior = qb;
        }
        public void quack()
        {
            Console.WriteLine("(I am a duck caller)");
            QuackBehavior.quack();
        }
    }
}